dessia_common/core.py

```diff
@@ -182,138 +164,45 @@ def schema(cls):
-            ordered_attributes = cls._ordered_attributes
-        else:
-            ordered_attributes = list(annotations.keys())
-
-        unordered_count = 0
-
-        # Parse docstring
-        try:
-            docstring = cls.__doc__
-            parsed_docstring = dcs.parse_docstring(docstring=docstring, annotations=annotations)
-        except Exception:
-            parsed_docstring = dcs.FAILED_DOCSTRING_PARSING
-        parsed_attributes = parsed_docstring['attributes']
-
-        # Initialize jsonschema
-        _jsonschema = deepcopy(JSONSCHEMA_HEADER)
-
-        required_arguments, default_arguments = dcs.inspect_arguments(method=init, merge=False)
-        _jsonschema['required'] = required_arguments
-        _jsonschema['standalone_in_db'] = cls._standalone_in_db
-        _jsonschema['description'] = parsed_docstring['description']
-        _jsonschema['python_typing'] = str(cls)
-
-        # Set jsonschema
-        for annotation in annotations.items():
-            name = annotation[0]
-            if name in ordered_attributes:
-                order = ordered_attributes.index(name)
-            else:
-                order = len(ordered_attributes) + unordered_count
-                unordered_count += 1
-            if name in cls._titled_attributes:
-                title = cls._titled_attributes[name]
-            else:
-                title = None
-
-            if name != 'return':
-                editable = name not in cls._non_editable_attributes
-                annotation_type = type_from_annotation(annotation[1], cls)
-                annotation = (name, annotation_type)
-                jss_elt = jsonschema_from_annotation(annotation=annotation, jsonschema_element={}, order=order,
-                                                     editable=editable, title=title,
-                                                     parsed_attributes=parsed_attributes)
-                _jsonschema['properties'].update(jss_elt)
-                if name in default_arguments:
-                    default = set_default_value(_jsonschema["properties"], name, default_arguments[name])
-                    _jsonschema['properties'].update(default)
-
-        _jsonschema['classes'] = [cls.__module__ + '.' + cls.__name__]
-        _jsonschema['whitelist_attributes'] = cls._whitelist_attributes
-        return _jsonschema
+        warnings.warn("'jsonschema' method is deprecated. Use schema instead", DeprecationWarning)
+        return cls.schema()
 
     @property
-    def method_schemas(self):  # TODO This should be a classmethod, but is a property to avoid collision with workflow.
+    def raw_method_schemas(self):
         """ Generate dynamic schemas for methods of class. """
+        # TODO This should be a classmethod, but is a property to avoid collision with workflow.
         cls = self.__class__
         valid_method_names = [m for m in dir(cls) if not m.startswith('_') and m in cls._allowed_methods]
```

Comment:
Are you sure _allowed_methods are always specified ? I guess, if not, all methods are allowed unless they have a specific pattern